### PR TITLE
Add env var to export a hint file to optimize ASAR ordering

### PR DESCRIPTION
**Pull request by bengotow**
Overview here: https://github.com/atom/atom/issues/10163

### CODE REVIEW

File: docs/api/environment-variables.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ Starts the process as a normal Node.js process.
 
 Prints Chrome's internal logging to console.
 
+## `ELECTRON_LOG_ASAR_READS`
+
+When Electron reads from an ASAR file, log the the read offset and file path
```
Comment (zeke) [inline docs/api/environment-variables.md]: typo: the the

File: docs/api/environment-variables.md [hunk 2]
```diff
@@ -25,6 +25,12 @@ Starts the process as a normal Node.js process.
...
+to the system `tmpdir`. Th resulting file can be provided to the ASAR module
```
Comment (zeke) [inline docs/api/environment-variables.md]: typo: Th

**Comment by etiktin** [comment]
Some asar resources are loaded from the c++ side ([example](https://github.com/atom/electron/blob/c1673eb666540b2891794bcb2f477f4b0cee78de/atom/common/api/atom_api_native_image.cc#L137)). Will this code log them as well?

**Comment by bengotow** [comment]
Hey folks thanks for the feedback—sorry I dropped the ball on this one. I'll address those typos! @etiktin, this isn't currently logging resources loaded from the C++ side. I'm not sure if it's possible to log those to the same file, and ensure that they're written to the logged in the correct order...

**Comment by zcbenz** [comment]
@bengotow Do you mind rebasing this on master branch? Recently we have converted all our CoffeeScript files to JavaScript.

**Comment by bengotow** [comment]
Glad to see everything is going to ES6. I've converted my changes to JS and updated the pull request. Also double-checked to make sure it still works correctly.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**